src/nodes.coffee

```diff
@@ -2519,7 +2564,7 @@ exports.ExportSpecifier = class ExportSpecifier extends ModuleSpecifier
 exports.Assign = class Assign extends Base
   constructor: (@variable, @value, @context, options = {}) ->
     super()
-    {@param, @subpattern, @operatorToken, @moduleDeclaration, @originalContext = @context} = options
+    {@param, @subpattern, @operatorToken, @moduleDeclaration, @shorthand, @originalContext = @context} = options
```

Comment:
This "making `Assign` more self-aware of whether/how it's an object property" via `@shorthand` would also seem to be evidence of a missing abstraction